### PR TITLE
make it easier to add support for other providers

### PR DESCRIPTION
**Pull request by srinivasupadhya**

### CODE REVIEW

File: src/main/java/in/ashwanthkumar/gocd/github/provider/github/model/PullRequestStatus.java [hunk 1]
```diff
@@ -10,25 +9,18 @@
     private String lastHead;
     // We use this to find changes in a PR
     private String mergeSHA;
-    @Exclude
```
Comment (ashwanthkumar) [inline src/main/java/in/ashwanthkumar/gocd/github/provider/github/model/PullRequestStatus.java]: Do you think we need these fields on the payload we store about the PR?

Comment (srinivasupadhya) [inline src/main/java/in/ashwanthkumar/gocd/github/provider/github/model/PullRequestStatus.java]: After #14 we only store {"pr-number": "latest-SHA"}. Its given as `Map` by git-cmd & stored as is.

Comment (ashwanthkumar) [inline src/main/java/in/ashwanthkumar/gocd/github/provider/github/model/PullRequestStatus.java]: Ah right, cool then.

**Comment by ashwanthkumar** [comment]
This is super nice.

**Comment by srinivasupadhya** [comment]
yes. makes sense. we might have to rename the project as well? not sure about that.

**Comment by srinivasupadhya** [comment]
This looks good to be merged. Did a sanity on git feature branch support & stash. Seems to be working. https://github.com/gocd/gocd/pull/940 needs to go in for everything to work.

To create `github` plugin:
```
mvn clean install -DskipTests
```
To create `git feature branch` plugin:
```
mvn clean install -DskipTests -P git.fb
```
To create `stash` plugin:
```
mvn clean install -DskipTests -P stash.pr
```
It should be fairly straight forward to implement support for other which is based on fetch refspec & prune by pattern approach.

**Comment by srinivasupadhya** [comment]
@ashwanthkumar im done with all the changes i intended to do. take a look & accept it if all looks fine.
After the PR is merged we will need to rename the project, change package (remove github as you mentioned) & update links [here](http://www.go.cd/community/plugins.html)

**Comment by srinivasupadhya** [comment]
you will now be able to run `release.sh` to create plugin jars for all plugins. they will be able put in `dist` directory & you can upload all its contents in a github release.

**Comment by ashwanthkumar** [comment]
Thanks for this. Any suggestions on how / to what we can rename the project?

**Comment by srinivasupadhya** [comment]
no. i thought about it but couldn't come up with a good name.

**Comment by ashwanthkumar** [comment]
How about `gocd-pr-plugin`? Also how do you suggest we should do this - I go ahead and create one or do you want take it up into gocd org?

**Comment by srinivasupadhya** [comment]
i was thinking `gocd-branch-change-poller`. effectively its just that. even the PRs are treated as anyother branch. we fetch updates for all branches & look for changes in a branch. Not sure if its a good name though.

**Pull request merged**